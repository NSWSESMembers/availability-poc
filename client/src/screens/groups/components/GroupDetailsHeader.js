import React from 'react';
import { View, Text } from 'react-native';
import PropTypes from 'prop-types';

import styles from './styles';
import IconButton from './IconButton';
import Icon from '../../../components/Icon';


const GroupDetailsHeader = ({
  groupIcon,
  groupName,
  tags,
  memberAlready,
  leaveGroup,
  joinGroup,
}) => (
  <View style={styles.topContainer}>
    <View style={styles.iconLeftHolder}>
      <Icon style={styles.icon} name={groupIcon} size={30} />
    </View>
    <View style={styles.textContainer}>
      <Text style={styles.titleTextBold} numberOfLines={1}>
        {groupName}
      </Text>
      <Text style={styles.subtitleText} numberOfLines={1}>
        {tags !== '' ? tags : 'No Tags'}
      </Text>
    </View>
    <IconButton
      onPress={memberAlready ? leaveGroup : joinGroup}
      icon={memberAlready ? 'fa-sign-out' : 'fa-sign-in'}
      text={memberAlready ? 'Leave Group' : 'Join Group'}
    />
  </View>
);

GroupDetailsHeader.propTypes = {
  groupIcon: PropTypes.string.isRequired,
  groupName: PropTypes.string.isRequired,
  tags: PropTypes.string.isRequired,
  memberAlready: PropTypes.bool.isRequired,
  leaveGroup: PropTypes.func.isRequired,
  joinGroup: PropTypes.func.isRequired,
};

export default GroupDetailsHeader;
