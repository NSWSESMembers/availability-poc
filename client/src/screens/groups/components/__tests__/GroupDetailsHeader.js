import React from 'react';
import renderer from 'react-test-renderer';

import GroupDetailsHeader from '../GroupDetailsHeader';

const noop = () => {};

test('renders correctly', () => {
  const tree = renderer.create(
    <GroupDetailsHeader
      groupIcon="fa-group"
      groupName="groupName"
      tags="tags"
      memberAlready
      leaveGroup={noop}
      joinGroup={noop}
    />,
  ).toJSON();
  expect(tree).toMatchSnapshot();

  const tree2 = renderer.create(
    <GroupDetailsHeader
      groupIcon="fa-group"
      groupName="groupName"
      tags=""
      memberAlready
      leaveGroup={noop}
      joinGroup={noop}
    />,
  ).toJSON();
  expect(tree2).toMatchSnapshot();

  const tree3 = renderer.create(
    <GroupDetailsHeader
      groupIcon="fa-group"
      groupName="groupName"
      tags="tags"
      memberAlready={false}
      leaveGroup={noop}
      joinGroup={noop}
    />,
  ).toJSON();
  expect(tree3).toMatchSnapshot();
});
