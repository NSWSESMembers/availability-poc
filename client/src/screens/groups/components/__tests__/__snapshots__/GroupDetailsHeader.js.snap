// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders correctly 1`] = `
<View
  style={
    Object {
      "backgroundColor": "white",
      "borderRadius": 2,
      "elevation": 1,
      "flexDirection": "row",
      "justifyContent": "space-between",
      "marginBottom": 0,
      "marginLeft": 10,
      "marginRight": 10,
      "marginTop": 6,
      "paddingBottom": 10,
      "paddingLeft": 8,
      "paddingRight": 8,
      "paddingTop": 10,
    }
  }
>
  <View
    style={
      Object {
        "alignItems": "center",
        "flexDirection": "row",
        "justifyContent": "center",
        "paddingRight": 8,
        "width": 40,
      }
    }
  >
    <Text
      accessible={true}
      allowFontScaling={false}
      ellipsizeMode="tail"
      style={
        Array [
          Object {
            "color": undefined,
            "fontSize": 30,
          },
          Object {
            "alignSelf": "center",
          },
          Object {
            "fontFamily": "FontAwesome",
            "fontStyle": "normal",
            "fontWeight": "normal",
          },
        ]
      }
    >
      
    </Text>
  </View>
  <View
    style={
      Object {
        "flex": 1,
        "flexDirection": "column",
      }
    }
  >
    <Text
      accessible={true}
      allowFontScaling={true}
      ellipsizeMode="tail"
      numberOfLines={1}
      style={
        Object {
          "flex": 0,
          "fontWeight": "bold",
          "includeFontPadding": false,
          "marginRight": 8,
          "textAlignVertical": "top",
        }
      }
    >
      groupName
    </Text>
    <Text
      accessible={true}
      allowFontScaling={true}
      ellipsizeMode="tail"
      numberOfLines={1}
      style={
        Object {
          "color": "#6B6B6B",
          "flex": 0,
          "fontSize": 12,
          "includeFontPadding": false,
          "textAlignVertical": "bottom",
        }
      }
    >
      tags
    </Text>
  </View>
  <View
    accessibilityComponentType={undefined}
    accessibilityLabel={undefined}
    accessibilityTraits={undefined}
    accessible={true}
    collapsable={undefined}
    hasTVPreferredFocus={undefined}
    hitSlop={undefined}
    isTVSelectable={true}
    nativeID={undefined}
    onLayout={undefined}
    onResponderGrant={[Function]}
    onResponderMove={[Function]}
    onResponderRelease={[Function]}
    onResponderTerminate={[Function]}
    onResponderTerminationRequest={[Function]}
    onStartShouldSetResponder={[Function]}
    style={
      Object {
        "alignItems": "center",
        "flexDirection": "column",
        "justifyContent": "center",
        "opacity": 1,
        "paddingLeft": 8,
        "width": 50,
      }
    }
    testID={undefined}
    tvParallaxProperties={undefined}
  >
    <Text
      accessible={true}
      allowFontScaling={false}
      ellipsizeMode="tail"
      style={
        Array [
          Object {
            "color": undefined,
            "fontSize": 25,
          },
          Object {
            "alignSelf": "center",
          },
          Object {
            "fontFamily": "FontAwesome",
            "fontStyle": "normal",
            "fontWeight": "normal",
          },
        ]
      }
    >
      
    </Text>
    <Text
      accessible={true}
      allowFontScaling={true}
      ellipsizeMode="tail"
      style={
        Object {
          "alignItems": "center",
          "color": "#6B6B6B",
          "flex": 0,
          "fontSize": 7,
          "includeFontPadding": false,
          "justifyContent": "center",
          "textAlign": "center",
        }
      }
    >
      Leave Group
    </Text>
  </View>
</View>
`;

exports[`renders correctly 2`] = `
<View
  style={
    Object {
      "backgroundColor": "white",
      "borderRadius": 2,
      "elevation": 1,
      "flexDirection": "row",
      "justifyContent": "space-between",
      "marginBottom": 0,
      "marginLeft": 10,
      "marginRight": 10,
      "marginTop": 6,
      "paddingBottom": 10,
      "paddingLeft": 8,
      "paddingRight": 8,
      "paddingTop": 10,
    }
  }
>
  <View
    style={
      Object {
        "alignItems": "center",
        "flexDirection": "row",
        "justifyContent": "center",
        "paddingRight": 8,
        "width": 40,
      }
    }
  >
    <Text
      accessible={true}
      allowFontScaling={false}
      ellipsizeMode="tail"
      style={
        Array [
          Object {
            "color": undefined,
            "fontSize": 30,
          },
          Object {
            "alignSelf": "center",
          },
          Object {
            "fontFamily": "FontAwesome",
            "fontStyle": "normal",
            "fontWeight": "normal",
          },
        ]
      }
    >
      
    </Text>
  </View>
  <View
    style={
      Object {
        "flex": 1,
        "flexDirection": "column",
      }
    }
  >
    <Text
      accessible={true}
      allowFontScaling={true}
      ellipsizeMode="tail"
      numberOfLines={1}
      style={
        Object {
          "flex": 0,
          "fontWeight": "bold",
          "includeFontPadding": false,
          "marginRight": 8,
          "textAlignVertical": "top",
        }
      }
    >
      groupName
    </Text>
    <Text
      accessible={true}
      allowFontScaling={true}
      ellipsizeMode="tail"
      numberOfLines={1}
      style={
        Object {
          "color": "#6B6B6B",
          "flex": 0,
          "fontSize": 12,
          "includeFontPadding": false,
          "textAlignVertical": "bottom",
        }
      }
    >
      No Tags
    </Text>
  </View>
  <View
    accessibilityComponentType={undefined}
    accessibilityLabel={undefined}
    accessibilityTraits={undefined}
    accessible={true}
    collapsable={undefined}
    hasTVPreferredFocus={undefined}
    hitSlop={undefined}
    isTVSelectable={true}
    nativeID={undefined}
    onLayout={undefined}
    onResponderGrant={[Function]}
    onResponderMove={[Function]}
    onResponderRelease={[Function]}
    onResponderTerminate={[Function]}
    onResponderTerminationRequest={[Function]}
    onStartShouldSetResponder={[Function]}
    style={
      Object {
        "alignItems": "center",
        "flexDirection": "column",
        "justifyContent": "center",
        "opacity": 1,
        "paddingLeft": 8,
        "width": 50,
      }
    }
    testID={undefined}
    tvParallaxProperties={undefined}
  >
    <Text
      accessible={true}
      allowFontScaling={false}
      ellipsizeMode="tail"
      style={
        Array [
          Object {
            "color": undefined,
            "fontSize": 25,
          },
          Object {
            "alignSelf": "center",
          },
          Object {
            "fontFamily": "FontAwesome",
            "fontStyle": "normal",
            "fontWeight": "normal",
          },
        ]
      }
    >
      
    </Text>
    <Text
      accessible={true}
      allowFontScaling={true}
      ellipsizeMode="tail"
      style={
        Object {
          "alignItems": "center",
          "color": "#6B6B6B",
          "flex": 0,
          "fontSize": 7,
          "includeFontPadding": false,
          "justifyContent": "center",
          "textAlign": "center",
        }
      }
    >
      Leave Group
    </Text>
  </View>
</View>
`;

exports[`renders correctly 3`] = `
<View
  style={
    Object {
      "backgroundColor": "white",
      "borderRadius": 2,
      "elevation": 1,
      "flexDirection": "row",
      "justifyContent": "space-between",
      "marginBottom": 0,
      "marginLeft": 10,
      "marginRight": 10,
      "marginTop": 6,
      "paddingBottom": 10,
      "paddingLeft": 8,
      "paddingRight": 8,
      "paddingTop": 10,
    }
  }
>
  <View
    style={
      Object {
        "alignItems": "center",
        "flexDirection": "row",
        "justifyContent": "center",
        "paddingRight": 8,
        "width": 40,
      }
    }
  >
    <Text
      accessible={true}
      allowFontScaling={false}
      ellipsizeMode="tail"
      style={
        Array [
          Object {
            "color": undefined,
            "fontSize": 30,
          },
          Object {
            "alignSelf": "center",
          },
          Object {
            "fontFamily": "FontAwesome",
            "fontStyle": "normal",
            "fontWeight": "normal",
          },
        ]
      }
    >
      
    </Text>
  </View>
  <View
    style={
      Object {
        "flex": 1,
        "flexDirection": "column",
      }
    }
  >
    <Text
      accessible={true}
      allowFontScaling={true}
      ellipsizeMode="tail"
      numberOfLines={1}
      style={
        Object {
          "flex": 0,
          "fontWeight": "bold",
          "includeFontPadding": false,
          "marginRight": 8,
          "textAlignVertical": "top",
        }
      }
    >
      groupName
    </Text>
    <Text
      accessible={true}
      allowFontScaling={true}
      ellipsizeMode="tail"
      numberOfLines={1}
      style={
        Object {
          "color": "#6B6B6B",
          "flex": 0,
          "fontSize": 12,
          "includeFontPadding": false,
          "textAlignVertical": "bottom",
        }
      }
    >
      tags
    </Text>
  </View>
  <View
    accessibilityComponentType={undefined}
    accessibilityLabel={undefined}
    accessibilityTraits={undefined}
    accessible={true}
    collapsable={undefined}
    hasTVPreferredFocus={undefined}
    hitSlop={undefined}
    isTVSelectable={true}
    nativeID={undefined}
    onLayout={undefined}
    onResponderGrant={[Function]}
    onResponderMove={[Function]}
    onResponderRelease={[Function]}
    onResponderTerminate={[Function]}
    onResponderTerminationRequest={[Function]}
    onStartShouldSetResponder={[Function]}
    style={
      Object {
        "alignItems": "center",
        "flexDirection": "column",
        "justifyContent": "center",
        "opacity": 1,
        "paddingLeft": 8,
        "width": 50,
      }
    }
    testID={undefined}
    tvParallaxProperties={undefined}
  >
    <Text
      accessible={true}
      allowFontScaling={false}
      ellipsizeMode="tail"
      style={
        Array [
          Object {
            "color": undefined,
            "fontSize": 25,
          },
          Object {
            "alignSelf": "center",
          },
          Object {
            "fontFamily": "FontAwesome",
            "fontStyle": "normal",
            "fontWeight": "normal",
          },
        ]
      }
    >
      
    </Text>
    <Text
      accessible={true}
      allowFontScaling={true}
      ellipsizeMode="tail"
      style={
        Object {
          "alignItems": "center",
          "color": "#6B6B6B",
          "flex": 0,
          "fontSize": 7,
          "includeFontPadding": false,
          "justifyContent": "center",
          "textAlign": "center",
        }
      }
    >
      Join Group
    </Text>
  </View>
</View>
`;
